js/adapt-pageIncompletePrompt.js

```diff
@@ -1,183 +1,174 @@
-        _prompts: [{
-          promptText: this.model._buttons.yes,
-          _callbackEvent: 'pageIncompletePrompt:leavePage'
-        }, {
-          promptText: this.model._buttons.no,
-          _callbackEvent: 'pageIncompletePrompt:cancel'
-        }],
-        _showIcon: true
-      };
-
-      // override with page-specific settings?
-      const pipConfig = this.pageModel.get('_pageIncompletePrompt');
-      if (pipConfig && pipConfig._buttons) {
-        promptObject.title = pipConfig.title;
-        promptObject.body = pipConfig.message;
-        promptObject._classes = pipConfig._classes;
-        promptObject._prompts[0].promptText = pipConfig._buttons.yes;
-        promptObject._prompts[1].promptText = pipConfig._buttons.no;
-      }
-
-      this.listenToOnce(Adapt, 'notify:cancelled', this.onLeaveCancel);
-      Adapt.trigger('notify:prompt', promptObject);
-      this.inPopup = true;
-    },
-
-    isEnabled: function() {
-      if (!Adapt.location._currentId) return false;
-      if (!this.handleRoute) return false;
-      if (!this.inPage) return false;
-      if (this.inPopup) return false;
-      if (this.isChangingLanguage) return false;
-
-      switch (Adapt.location._contentType) {
-        case 'menu': case 'course':
-          this.inPage = false;
-          return false;
-      }
-
-      const pageModel = Adapt.findById(Adapt.location._currentId);
-      if (pageModel.get('_isOptional')) return false;
-
-      const isEnabledForCourse = this.model && !!this.model._isEnabled;
-      const isEnabledForPage = pageModel.get('_pageIncompletePrompt') && !!pageModel.get('_pageIncompletePrompt')._isEnabled;
-      return (isEnabledForCourse && isEnabledForPage !== false) || isEnabledForPage;
-    },
-
-    enableRouterNavigation: function(value) {
-      Adapt.router.set('_canNavigate', value, { pluginName: this.PLUGIN_NAME });
+import Adapt from 'core/js/adapt';
+import data from 'core/js/data';
+import device from 'core/js/device';
+import location from 'core/js/location';
+import router from 'core/js/router';
+
+class PageIncompletePrompt extends Backbone.Controller {
+  initialize() {
+    this.handleRoute = true;
+    this.inPage = false;
+    this.inPopup = false;
+    this.isChangingLanguage = false;
+    this.pageModel = null;
+    this._ignoreAccessibilityNavigation = false;
+
+    this.setupEventListeners();
+  }
+
+  setupEventListeners() {
+    _.bindAll(this, 'onLanguageChanging', 'onPageViewReady', 'onLeavePage', 'onLeaveCancel', 'onRouterNavigate');
+
+    this.listenTo(Adapt, {
+      'app:languageChanged': this.onLanguageChanging,
+      'pageView:ready': this.onPageViewReady,
+      'pageIncompletePrompt:leavePage': this.onLeavePage,
+      'pageIncompletePrompt:cancel': this.onLeaveCancel,
+      'router:navigate': this.onRouterNavigate
+    });
+
+    this.listenToOnce(Adapt, 'app:dataLoaded', function() {
+      this.listenTo(Adapt, 'accessibility:toggle', this.onAccessibilityToggle);
```

Comment:
It's odd that this is attached only after `app:dataLoaded`. 
I don't think `accessibility:toggle` exists now, as a11y is on all of the time and we don't now have a button to toggle its behaviour? All of this can be removed.

Old code: https://github.com/search?q=repo%3Aadaptlearning%2Fadapt-contrib-core+accessibility%3Atoggle&type=commits